### DISCUSSION TITLE
Automate/Play Back A Roast Using Saved Profile

### DISCUSSION THREAD

**Discussion by gevans3** [artisan-roaster-scope/artisan#1140] [Q&A]
Every time I look for how to automate a roast on my Hot Top 2K, I get directed back to the same post about creating backgrounds.  I don’t need that, this still requires the manual management/manipulation of a roast manually “over the top” of the profile.  What I’m looking for is a way to load a profile to duplicate the roast automatically exactly to the events/triggers/slider/burner/times values of a previously successful roast.  I’ve tried creating events in alarms to see if I can automate using alarms, but when I “create alarm” from the roast/events tab, it shows up in the config/alarms tab but all the events are “blank” with no values.  If I have 7 events in my profile/roast/events window, and I “create alarms” from them, 7 alarms will show up in config/alarms, but they are all generic “add alarm” rows that still require all the values and info to be manually inserted. Various tutorials and etc. suggest that alarms can be created from the events in a roast profile and subsequent roasts can be effectively automated/replicated.  But I can’t seem to get it to work. Don’t know if it’s a bug or simply inelegant UX.  Please advise if you have successfully automated/playback subsequent roasts from an existing profile. Thank you.

**Comment by poundy** [comment]
first, best to include your settings file here (don't reply to email and attach it, save a copy from Artisan, rename it as a TXT extension file, and drop it in a reply box here).  You will need to do the same with an example background roast that you want to automate.

You should also explain from that setting file what your approach has been to enable it to replicate the background.

In my (limited) exposure, I'd suggest that 100% of replicating roasts is done with PID, which is intending to follow a roast profile's path not through the steps taken once to achieve a goal, but by actively managing bean temp. A simple change in exhaust back pressure through chaff buildup, for example, could be sufficient to cause your simple "step" approach to replicating a roast to not achieve the same roast profile.

**Comment by MAKOMO** [comment]
The HOTTOP heat management is too slow to successfully duplicate a roast using a PID. Why not just load a successful profile in the background and tick the "Playback Events" in the Background dialog (menu Roast >> Background) by either time or temperature?

**Reply by MAKOMO** [reply]
Btw. the Create Alarm function under Roast >> Events works for me as expected. If you think it is broken it would help to get your settings and the profile you want to create the alarms from to investigate why this is not working for you.

**Comment by gevans3** [comment]
Thank you! I’ll double check my settings…it may just be that I don’t have
my preferences set properly for time/temperature, which could be why when I
DO bring up a background, the proper number of events show, but all of the
values are all at 0.00.

To your earlier point, manually roasting against a background roast is
exactly how I replicate my roasts (and to a great level of success now that
I’ve learned how to anticipate the manipulations on the Hot Top).  Indeed,
it is slow to react so I’ve learned to make any tweaks in advance of the
background telling me I need to.

I’ve gotten my FC to within around 10 seconds roast to roast and depending
upon the beans, within about 3 tenths of a degree of one another, my
development % within a half percent roast to roast and my drop temperatures
generally within a tenth of a degree or so…just did that with 6 roasts of a
Kenyan Kiranyagi last weekend and they turned out great.

I’ve been doing roasting “classes” with Rob Hoos at iteration.coffee and
his method focuses on time to first crack, development time and roast
color, with ROR secondary to all of these.  Even with that, my curves are
pretty nice.

Long story short, I’m going to try what you suggest with the event playback
if I can get it to work, but if not, I’m not going to sweat it.  Can’t
thank you enough for your kind attention.

Geo

On Tue, Apr 18, 2023 at 11:29 AM Marko Luther ***@***.***>
wrote:

> The HOTTOP heat management is too slow to successfully duplicate a roast
> using a PID. Why not just load a successful profile in the background and
> tick the "Playback Events" in the Background dialog (menu Roast >>
> Background) by either time or temperature?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/artisan-roaster-scope/artisan/discussions/1140#discussioncomment-5650606>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABZPQAJHKCKDCJ5JSYWFVDTXB2XL7ANCNFSM6AAAAAAWQKI7W4>
> .
> You are receiving this because you authored the thread.Message ID:
> <artisan-roaster-scope/artisan/repo-discussions/1140/comments/5650606@
> github.com>
>
-- 


George Evans
317.869.3438

*"The completely un-ambitious writer and the really good unpublished poem
are the things we lack most at this time. There is, of course, the problem
of sustenance."*
*
                       -- Evan Shipman*
